### PR TITLE
Some fixes for Dragonflight

### PR DESCRIPTION
**Pull request by codegain**
Hi,

I did some digging in the code and fixed Carbonite Map to the point that it is useable without errors again. No special settings tested and still needs some work probably.

### CODE REVIEW

File: NxMap.lua [hunk 1]
```diff
@@ -678,7 +678,7 @@ function Nx.Map:Create (index)
 	f:SetFrameStrata ("LOW")
 	f:SetWidth (m.W)
 	f:SetHeight (m.H)
-	f:SetMinResize (50, 50)
+	f:SetResizeBounds (50, 50)
```
Comment (codegain) [inline NxMap.lua:681]: `SetMinResize` has been removed and replaced by `SetResizeBounds`, see https://wowpedia.fandom.com/wiki/API_Frame_SetMinResize

File: NxMap.lua [hunk 2]
```diff
@@ -1321,7 +1321,7 @@ function Nx.Map:CreateLocationTip()
 --	f.NxInst = self
 	self.LocTipFrm = f
 
-	f:SetClampedToScreen()
+	f:SetClampedToScreen(true)
```
Comment (codegain) [inline NxMap.lua:1324]: This now needs an argument and apparently has no default value anymore

File: NxMap.lua [hunk 3]
```diff
@@ -5068,7 +5068,7 @@ function Nx.Map:Update (elapsed)
 							f.texture:SetAtlas(atlasIcon)
 						else
 							f.texture:SetTexture ("Interface\\Minimap\\POIIcons")
-							txX1, txX2, txY1, txY2 = GetPOITextureCoords (txIndex)
+							txX1, txX2, txY1, txY2 = C_Minimap.GetPOITextureCoords (txIndex)
```
Comment (codegain) [inline NxMap.lua:5071]: Moved to `C_Minimap`, see https://wowpedia.fandom.com/wiki/Patch_10.0.0/API_changes

File: NxUI.lua [hunk 4]
```diff
@@ -3456,7 +3456,7 @@ function Nx.Button:Init()
 	f:Hide()
 
 	local t = f:CreateTexture()
-	t:SetColorTexture (Nx.Util_str2rgba ("0.06|0.06|0.250|1"))
+	t:SetColorTexture (Nx.Util_str2rgb ("0.06|0.06|0.250"))
```
Comment (codegain) [inline NxUI.lua:3459]: I don't know what the bug here is - apparently `SetColorTexture` still accepts an alpha value, but this is not working anymore. I spent like 2.5 hours debugging this and even tested this code:

```
local r, g, b, a = Nx.Util_str2rgba ("0.06|0.06|0.250|1")
t:SetColorTexture(r, g, b, a)
```

which doesn't work (error like in #223).
I printed out the values of `r,g,b,a` and they were
```
0.06
0.06
0.250
1
```
which is what I expected. Harcoding the values like
```
t:SetColorTexture(0.06, 0.06, 0.250, 1)
```
also works and
```
t:SetColorTexture(r, g, b, 1)
```
also.

It's only that the alpha value does not seem to be working if passed as a variable.
I also tried setting it to `1.0` and `0.5`: Hardcoded works, but not as a variable.

File: Carbonite.xml [hunk 5]
```diff
@@ -34,6 +35,7 @@
 			<AbsInset left="3" right="3" top="6" bottom="6" />
 		</BackgroundInsets>
 	</Backdrop>
+-->
```
Comment (codegain) [inline Carbonite.xml:38]: Found no replacement for this one. It throws `Unknown XML element "Backdrop"`

**Comment by rhodan412** [comment]
Beautiful thank you.  The only error I am receiving (I just have Carbonite Map loaded and not the Quest module) is:

1x Interface/AddOns/Carbonite/NxOptions.lua:4923: attempt to index global 'MainMenuBarArtFrame' (a nil value)
[string "@Interface/AddOns/Carbonite/NxOptions.lua"]:4923: in function `NXCmdGryphonsUpdate'
[string "@Interface/AddOns/Carbonite/NxOptions.lua"]:4786: in function `func'
[string "@Interface/AddOns/Ace3/AceTimer-3.0-17/AceTimer-3.0.lua"]:57: in function <Interface/AddOns/Ace3/AceTimer-3.0/AceTimer-3.0.lua:50>

Locals:
self = <table> {
 NXCmdReload = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5138
 NXCmdImportCarbMine = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4970
 NXCmdMMButUpdate = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5017
 NXCmdImportCarbHerb = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4962
 GetVar = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5555
 QuickOptsTimer = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4795
 NXCmdImportCharSettings = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5046
 NXCmdGryphonsUpdate = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4921
 Skins = <table> {
 }
 NXCmdDeleteTimber = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4939
 NXCmdUIChange = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5154
 ChoicesQArea = <table> {
 }
 SetVar = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5579
 NXCmdMapToolBarUpdate = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5029
 Open = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4834
 UpdateCom = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5318
 ChoicesAnchor0 = <table> {
 }
 NXCmdResetWatchWinLayout = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5134
 NXCmdResetOpts = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5106
 Opts = <table> {
 }
 COpts = <table> {
 }
 ChoicesQAreaTex = <table> {
 }
 InitTimer = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4780
 EditCHAccept = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5381
 CalcChoices = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5444
 EditItem = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5341
 NXCmdDeleteCharSettings = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5075
 ChoicesAnchor = <table> {
 }
 OnSetSize = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5160
 NXCmdInfoWinUpdate = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5004
 Update = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5227
 AddToProfileMenu = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:55
 NXCmdImportCarbMisc = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4978
 NXCmdDeleteMine = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4946
 NXCmdHUDChange = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5147
 OnListEvent = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5173
 NXCmdFavCartImport = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4904
 ChoicesCorner = <table> {
 }
 NXCmdMMOwnChange = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5010
 ParseVar = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5507
 EditFAccept = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5394
 EditIAccept = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5410
 Init = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4739
 OnPageListEvent = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5165
 NXCmdQWHideRaid = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5042
 EditSAccept = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5426
 NXCmdFontChange = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4908
 NXCmdDeleteHerb = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4931
 NXCmdResetWinLayouts = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5125
 NXCmdDeleteMisc = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4954
 NXCmdQWFadeAll = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5038
 Create = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4842
 NXCmdMMChange = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:5024
 NXCmdCamForceMaxDist = <function> defined @Interface/AddOns/Carbonite/NxOptions.lua:4912
 Reset = <

**Pull request merged**